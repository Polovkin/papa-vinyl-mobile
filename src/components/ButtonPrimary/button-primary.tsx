import React, {FC} from 'react';
import {TouchableOpacityProps, Text, TouchableOpacity} from 'react-native';
import cn from 'classnames';

export enum ColorType {
  primary = 'primary',
  secondary = 'secondary',
}

interface Props extends TouchableOpacityProps {
  text: string;
  colorType?: ColorType;
}

const ButtonPrimary: FC<Props> = props => {
  const {text, colorType = ColorType.primary, className} = props;

  let typeStyle: string;
  switch (colorType) {
    case ColorType.primary:
      typeStyle = 'bg-primary';
      break;
    case ColorType.secondary:
      typeStyle = 'bg-secondary';
      break;
    default:
      typeStyle = 'bg-secondary';
  }
  console.log(colorType);
  return (
    <TouchableOpacity
      testID={'button-primary'}
      className={cn(
        'm-4 m-1 flex w-32 items-center justify-center rounded-xl py-2 px-4',
        typeStyle,
        className,
      )}>
      <Text className={'capitalize text-white'}>{text}</Text>
    </TouchableOpacity>
  );
};

export default ButtonPrimary;
