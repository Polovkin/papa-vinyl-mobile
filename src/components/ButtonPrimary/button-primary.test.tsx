import React from 'react';
import ButtonPrimary from './button-primary';
import {render, fireEvent} from '@testing-library/react-native';

describe('ButtonPrimary', () => {
  it('renders text', () => {
    const {getByText} = render(<ButtonPrimary text={'button'} />);
    const text = getByText('button');
    expect(text.props.children).toBe('button');
  });

  it('handles onPress', () => {
    const onPress = jest.fn();
    const {getByTestId} = render(
      <ButtonPrimary text={'button'} onPress={onPress} />,
    );
    fireEvent.press(getByTestId('button-primary'));
    expect(onPress).toHaveBeenCalled();
  });
});
