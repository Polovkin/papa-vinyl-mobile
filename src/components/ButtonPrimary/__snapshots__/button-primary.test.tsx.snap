// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimary to match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#c25dd7",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "display": "flex",
      "justifyContent": "center",
      "marginBottom": 4,
      "marginLeft": 4,
      "marginRight": 4,
      "marginTop": 4,
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 8,
      "width": 128,
    }
  }
  testID="button-primary"
>
  <Text
    style={
      [
        {
          "color": "#fff",
        },
      ]
    }
  >
    button
  </Text>
</View>
`;
